data_override/data_override.F90

```diff
@@ -365,20 +365,21 @@ subroutine read_table(data_table)
     type(data_type)  :: data_entry
 
     logical               :: ongrid
+    logical               :: table_exists !< Flag indicating existence of data_table
     character(len=128)    :: region, region_type
 
     integer :: sunit
 
 !  Read coupler_table
-    open(newunit=iunit, file='data_table', action='READ', iostat=io_status)
-    if(io_status/=0) then
-      if(io_status==29) then
-        call mpp_error(NOTE, 'data_override_mod: File data_table does not exist.')
-      else
-        call mpp_error(FATAL, 'data_override_mod: Error in opening file data_table.')
-      endif
-    endif
+    inquire(file='data_table', EXIST=table_exists)
```

Comment:
will this work for a yaml also?

Reply:
right now that check is in the yaml_parser, because the yaml_parser has to return an integer file_id (I've hardcoded that to 999 in the yaml_parser). Data_override then checks for a 999 file_id return from the yaml_parser. Is the suggestion here to handle the absence of the data_table.yaml directly in data_override (i.e. not even call the parser)?

Reply:
If the parser can handle it, then let the parser do it's job.